src/components/Filter/Filter.tsx

```diff
@@ -0,0 +1,85 @@
+import {
+  ComponentProps,
+  forwardRef,
+  Children,
+  cloneElement,
+  isValidElement,
+  ReactNode,
+  Dispatch,
+  SetStateAction,
+} from 'react';
+import { getCommonProps, px } from '../../utils';
+import classnames from 'classnames';
+import FilterHeader, { FilterHeaderProps } from './FilterHeader';
+import { FilterValueProps } from './FilterValue';
+import Button from '../Button/Button';
+import { ButtonVariants } from '../Button/types';
+import ChevronNextIcon from '../../assets/chevronNext.svg?react';
+
+// You'll need to change the ComponentProps<"htmlelementname"> to match the top-level element of your component
+export interface FilterProps extends ComponentProps<'div'> {
+  children: ReactNode;
+
+  // If true, do not include bottom border. True by default, do not define if you do not wish to use separators.
+  isLast?: boolean;
+
+  // Setter for values to display in view all
+  setViewAllFilter?: Dispatch<SetStateAction<ReactNode[]>>;
+
+  // Number of values to display before truncating with view all button
+  viewAllLimit?: number;
+
+  // Whether this is a view all filter or not
+  viewingAll?: boolean;
+}
+/**
+ * ## Overview
+ *
+ * Component to display filtering dimensions
+ *
+ * [Figma Link](https://www.figma.com/design/OvBXAq48blO1r4qYbeBPjW/RW---Sale-Page-(PLP)?node-id=892-71019&node-type=frame&t=AsBDn4UgUEjNUnf7-0)
+ *
+ * [Storybook Link](https://phillips-seldon.netlify.app/?path=/docs/components-filter--overview)
+ */
+const Filter = forwardRef<HTMLDivElement, FilterProps>(
+  ({ className, children, isLast = true, viewAllLimit = 10, viewingAll = false, setViewAllFilter, ...props }, ref) => {
+    const { className: baseClassName, ...commonProps } = getCommonProps(props, 'Filter');
+    const childrenArray = Children.toArray(children);
+
+    return (
+      <div
+        {...commonProps}
+        className={classnames(baseClassName, className, {
+          [`${px}-has-separators`]: !isLast,
```

Comment:
not really clear, why do we need this separator?

Reply:
okay I see now, maybe we should `Separator` component for that, @scottdickerson wdyt?

Reply:
![image](https://github.com/user-attachments/assets/0cbb7420-19ab-4463-a764-d7db4d2d307b)

Reply:
When there are multiple filters, the design shows there is a separator between them. This Filter component, for the majority of cases I would think, would be implemented alongside other Filters. I created the FilterControl component to wrap and manage these.